modules/api/api.py

```diff
@@ -57,6 +57,7 @@ def __init__(self, app: FastAPI, queue_lock: Lock):
         self.app.add_api_route("/sdapi/v1/samplers", self.get_samplers, methods=["GET"], response_model=List[SamplerItem])
         self.app.add_api_route("/sdapi/v1/upscalers", self.get_upscalers, methods=["GET"], response_model=List[UpscalerItem])
         self.app.add_api_route("/sdapi/v1/sd-models", self.get_sd_models, methods=["GET"], response_model=List[SDModelItem])
+        self.app.add_api_route("/sdapi/v1/sd-models", self.get_sd_models, methods=["POST"])
```

Comment:
Shouldn't this point to `self.set_sd_models` ?

Reply:
Yeah looks like that's the issue. Is there anyway to make it where you can specify the model to use in the txt2img and img2img request?

Reply:
You can't, but you can set it in the config, but that is broken for now

Reply:
im going to be remaking this PR to fix the config instead - this should also allow for model reloading

Reply:
There is a override settings field in img2txt and img2img request where you can use specific settings (including using a particular hypernetwork and model) for that inference, then restoring to previous state. Unfortunately, due to long loading times for the model, it doesnt work for model switching, but it is trivial enough to make it load on change. I already have an open PR to enable the hypernetwork loading - though it has not been merged yet.

Reply:
> There is a override settings field in img2txt and img2img request where you can use specific settings (including using a particular hypernetwork and model) for that inference, then restoring to previous state. Unfortunately, due to long loading times for the model, it doesnt work for model switching, but it is trivial enough to make it load on change. I already have an open PR to enable the hypernetwork loading - though it has not been merged yet. #4120 You can see this PR to see how to add support for reloading model.

I tried overriding `sd_model_checkpoint` with the model `title`, `model_name` and `hash` but none seemed to work as the hash for each image came back with the default model hash.

Reply:
Yes as I said, this does not work now, but you can quickly patch to make it work, refer to my PR, just call the reload method for model

Reply:
Hmm, I've been thinking about this and I agree that it's probably best done as part of the generation request itself. Otherwise, a user that tries to generate with another model may end up switching the model for pending generations, unless the client has its own internal queuing mechanism (which other clients won't be able to share.)

Has anyone made a PR like #4120 for switching the model within the request itself?

Reply:
> Has anyone made a PR like https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/4120 for switching the model within the request itself?
I can work on it in early Dec.